starport/pkg/cosmoscmd/proxy.go

```diff
@@ -0,0 +1,65 @@
+package cosmoscmd
+
+import (
+	"path/filepath"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/spf13/cobra"
+	"github.com/tendermint/starport/starport/pkg/cosmosutil"
+	"github.com/tendermint/starport/starport/pkg/ctxticker"
+	"github.com/tendermint/starport/starport/pkg/gitpod"
+	"github.com/tendermint/starport/starport/pkg/xchisel"
+	"github.com/tendermint/starport/starport/services/network/networkchain"
+)
+
+const TunnelRerunDelay = 5 * time.Second
+
+func startProxyForTunneledPeers(clientCtx client.Context, cmd *cobra.Command) {
+	if cmd.Name() != "start" {
+		return
+	}
+	serverCtx := server.GetServerContextFromCmd(cmd)
+	ctx := cmd.Context()
+	if gitpod.IsOnGitpod() {
+		go func() {
+			_ = ctxticker.DoNow(ctx, TunnelRerunDelay, func() error {
```

Comment:
maybe we can return this error and log it

```go
err := ctxticker.DoNow(ctx, TunnelRerunDelay, func() error {
	serverCtx.Logger.Info("Starting chisel server", "port", xchisel.DefaultServerPort)
	err := xchisel.StartServer(ctx, xchisel.DefaultServerPort)
	if err != nil {
		serverCtx.Logger.Error(
			"Failed to start chisel server",
			"port", xchisel.DefaultServerPort,
			"reason", err.Error(),
		)
	}
	return nil
})
if err != nil {
	serverCtx.Logger.Error(
		"Failed to start ticker",
		"reason", err.Error(),
	)
}
```

Reply:
I think, DoNow never returns an error since on error, it's set to retry indefinitely until ctx is canceled.

Inside the DoNow's callback, err is being logged, you think this meets what we need already?

Reply:
The idea is DoNow will never return an error. Because in case if it does it will exit but I don't want it exits.